src/Assets/LightSkyDefense/Scripts/Waves/Wave.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections;
+using UnityEngine;
+
+
+public abstract class Wave : ScriptableObject
+{
+    public abstract IEnumerator Spawn();
+
+    [Tooltip("The Amount of times this wave will repeat")]
+    [Min(0)]
+    public int Repeat = 1;
```

Comment:
https://docs.unity3d.com/ScriptReference/Object.FindObjectsOfType.html

Reply:
Well it mainly exists to make creating waves easier. Honestly, since I don't know what the final wave editor will look like its unclear if it will be needed.